projects/kit/components/stepper/stepper.component.ts

```diff
@@ -98,9 +98,11 @@ export class TuiStepperComponent {
     }
 
     indexOf(step: HTMLElement): number {
-        return tuiGetOriginalArrayFromQueryList(this.steps).findIndex(
+        const index = tuiGetOriginalArrayFromQueryList(this.steps).findIndex(
             ({nativeElement}) => nativeElement === step,
         );
+
+        return index < 0 ? NaN : index;
```

Comment:
Can we add any screenshot test for this?

Reply:
https://github.com/taiga-family/taiga-ui/pull/9638#discussion_r1825912511